docs/docs/development/hardware-integration/shift-registers.md

```diff
@@ -0,0 +1,150 @@
+---
+title: Configuring Shift Registers
+sidebar_label: Shift Registers
+---
+
+Shift registers are the recommended method of adding additional GPIO pins to MCUs and boards, when a standard matrix results in an insufficient number of keys. They are recommended because they simultaneously have very low power consumption and are quite cheap. This page serves as a (brief) introduction to shift registers, how to use them in your design, and how to configure ZMK to use them correctly.
+
+:::note
+This page assumes that the part you are using has the code 74HC595. Other shift registers can work as well, but this is the most commonly used one.
+:::
+
+:::tip
+To understand how shift registers work, we recommend reading through ["How does the 74HC595 Shift Register work?"](https://lastminuteengineers.com/74hc595-shift-register-arduino-tutorial/#how-does-the-74hc595-shift-register-work).
+:::
+
+## Design Guidelines
+
+The shift register output pins should be connected to the output pins of a keyboard matrix, while the input pins should be connected directly to the MCU/board. This is to allow the input pins to trigger "interrupts" on the MCU/board, upon which it will begin scanning the keys. Using a shift register for input pins is also possible, but requires you to use a different part and will harm your battery life.
+
+Identifying the output pins of your keyboard matrix is easy - these are the pins connected to the [anodes of your diodes](https://learn.sparkfun.com/tutorials/diodes/all#ideal-diodes). You most likely will need to rearrange your matrix to have significantly more inputs than outputs.
+
+You will want to make sure that the data and clock pins of the shift register are connected to high frequency/SPI-capable pins on your MCU. Make sure that these lie on the same SPI bus for your microcontroller, if applicable (rp2040 yes, nRF52840 no). It is generally recommended that you use the pre-defined pins for SPI, if your board comes with them.
+
+For daisy chaining purposes, ZMK allows you to chain a maximum of four shift registers together.
```

Comment:
Is the hardware setup for daisy chaining in the scope of this document? Asking because e.g. I don't know how exactly and maybe a link or a remark would be nice.

Reply:
Added a schematic image.